### PR TITLE
fix(dashboard): make to filter the correct certified or non-certified…

### PR DESCRIPTION
**Pull request by prosdev0107**
### SUMMARY
Non certified dashboards appear on dashboard list when filtering by certified

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE:
In case of certified filter is Yes
image

In case of certified filter is No
image

AFTER:
In case of certified filter is Yes
image

In case of certified filter is No
image

### TESTING INSTRUCTIONS

**How to reproduce the bug**
1. Certify some dashboards (This can be done from the dashboard CRUD view > edit > certified fields at the bottom of the modal)
2. Go to the dashboard list
3. Should show only certificated Dashboards when the filter option of certified is Yes.

Generally, You will not reproduce this issue anymore with dashboard CRUD. This issue is happened when the certified_by field value is empty string, not null value.(like following image). and so I had a test after changing this forcely into empty string in some dashboards value and then could catch this issue and fix it.
image

Or you can see this issue on the following workspace;
https://8bf970d2.us1a.app.preset.io/dashboard/list/?filters=(certified:(label:Yes,value:!t))&pageIndex=0&sortColumn=changed_on_delta_humanized&sortOrder=desc&viewMode=table

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by diegomedina248 — APPROVED** [review]

**Review by rusackas — APPROVED** [review]
LGTM!

File: superset/dashboards/filters.py [hunk 1]
```diff
@@ -189,12 +189,14 @@ def apply(self, query: Query, value: Any) -> Query:
             return query.filter(
                 and_(
                     Dashboard.certified_by.isnot(None),
+                    Dashboard.certified_by != '',
```
Comment (rusackas) [inline superset/dashboards/filters.py:192]:
```suggestion
                    Dashboard.certified_by != "",
```

File: superset/dashboards/filters.py [hunk 2]
```diff
@@ -189,12 +189,14 @@ def apply(self, query: Query, value: Any) -> Query:
...
                 )
             )
         if value is False:
             return query.filter(
-                and_(
+                or_(
                     Dashboard.certified_by.is_(None),
+                    Dashboard.certified_by == '',
```
Comment (rusackas) [inline superset/dashboards/filters.py:199]:
```suggestion
                    Dashboard.certified_by == "",
```
These two changes should fix the `black` linting issues the PR is facing. 

FWIW, I looked a the CI logs, which show the file failing linting. You can then run `black superset/dashboards/filters.py` to fix/commit these changes.

**Review by rusackas — CHANGES_REQUESTED** [review]
Added a couple of click-to-commit suggestions that should fix the Python linting issues blocking this PR.

**Review by rusackas — APPROVED** [review]

**Review by zhaoyongjie — APPROVED** [review]
Thanks for fixing this issue. For the short term, the fix is good. for the long term, I suggest adding `todo` notation in the codebase. We might set the `certified_by` to `NULL` in the [Dashboard model](https://github.com/apache/superset/blob/a2d34ec4b8a89723e7468f194a98386699af0bd7/superset/models/dashboard.py#L145) when the user deactivates certified so that only keep `is null` in where clause.

**Comment by rusackas** [comment]
> Thanks for fixing this issue. For the short term, the fix is good. for the long term, I suggest adding `todo` notation in the codebase. We might set the `certified_by` to `NULL` in the [Dashboard model](https://github.com/apache/superset/blob/a2d34ec4b8a89723e7468f194a98386699af0bd7/superset/models/dashboard.py#L145) when the user deactivates certified so that only keep `is null` in where clause.

@prosdev0107 do you want to add that TODO as suggested?

**Comment by prosdev0107** [comment]
@rusackas @zhaoyongjie 
According to my local testing result, the certified_by is keeping null value when user creates new dashboard or deactivates existing certified on current codebase. So I am wondering if I should add this TODO.

**Pull request merged**